File: February2022/Java/Subset.java
//Author Shobhit Behl(LC:shobhitbruh)
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans=new ArrayList<>();
        List<Integer> al=new ArrayList<>();
        rec(al,ans,nums,nums.length-1);
        
        return ans;
    }
    
    public void rec(List<Integer> al,List<List<Integer>> ans,int[] nums,int i){
        if(i==-1){
            List<Integer> ak=new ArrayList<>();
            for(int j=0;j<al.size();j++){
                ak.add(al.get(j));
            }
            ans.add(ak);
            return;
        }
        
        rec(al,ans,nums,i-1);
        al.add(nums[i]);
        rec(al,ans,nums,i-1);
        al.remove(al.size()-1);
        
        
    }
}

File: February2022/Kotlin/subset.kt
//Author Shobhit Behl(LC:shobhitbruh)
class Solution {
    fun subsets(nums: IntArray): List<List<Int>> {
        val ans: MutableList<List<Int>> = ArrayList()
        val al: MutableList<Int> = ArrayList()
        rec(al, ans, nums, nums.size - 1)
        return ans
    }

    fun rec(al: MutableList<Int>, ans: MutableList<List<Int>>, nums: IntArray, i: Int) {
        if (i == -1) {
            val ak: MutableList<Int> = ArrayList()
            for (j in al.indices) {
                ak.add(al[j])
            }
            ans.add(ak)
            return
        }
        rec(al, ans, nums, i - 1)
        al.add(nums[i])
        rec(al, ans, nums, i - 1)
        al.removeAt(al.size - 1)
    }
}